### PR TITLE
Remove legacy role settings

### LINKED ISSUE #66409: Remove the legacy node role settings
In #54998 targeting 7.9.0, we introduced a new `node.roles` settings for configuring the roles that a node has. These supersede the legacy node settings such as `node.data`, `node.master`, etc., which were deprecated in the same release. For example, to configure a dedicated ML node, previously you would use `node.ml: true` and now you’d use `node.roles: [ “ml” ].` 

We intend to remove the legacy node role settings in the 8.0.0 release of Elasticsearch. This issue is a placeholder for following that work.

### PR DESCRIPTION
**Pull request by jasontedor**
This commit removes the previously deprecated legacy role settings. These settings have been replaced by node.roles.

Relates #54998
Relates #71143
Closes #66409

### CODE REVIEW

File: docs/reference/migration/migrate_8_0/settings.asciidoc [hunk 1]
```diff
@@ -185,3 +185,27 @@ If you would like to use wildcard patterns for destructive actions, set
...
+* `node.data`
+* `node.ingest`
+* `node.master`
+* `node.ml`
+* `node.remote_cluster_client`
+* `node.transform`
+* `node.voting_only`
+
+have been removed. Instead, use the `node.roles` setting. If you were previously
+using the legacy role settings on 7.13 or later cluster, you will have a
```
Comment (rjernst) [inline docs/reference/migration/migrate_8_0/settings.asciidoc:204]: nit: on **a** 7.13 or later cluster

File: docs/reference/migration/migrate_8_0/settings.asciidoc [hunk 2]
```diff
@@ -185,3 +185,27 @@ If you would like to use wildcard patterns for destructive actions, set
...
+deprecation log message on each of your node informing you of the exact
```
Comment (rjernst) [inline docs/reference/migration/migrate_8_0/settings.asciidoc:205]: Maybe reword the last part of this:

on each of your **nodes indicating the exact replacement value for `node.roles`.**

File: server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeRole.java [hunk 3]
```diff
@@ -20,7 +18,7 @@
 /**
  * Represents a node role.
  */
-public abstract class DiscoveryNodeRole implements Comparable<DiscoveryNodeRole> {
+public class DiscoveryNodeRole implements Comparable<DiscoveryNodeRole> {
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeRole.java:21]: This could be a followup, but I wonder if we could make this final now. Seems like the only need for override is the enabled by default predicate, but this could be a lambda passed to the second constructor?

**Review by rjernst — APPROVED** [review]
LGTM. Just a couple minor suggestions

File: server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeRole.java (hunk 3 above)
Comment (jasontedor) [reply server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeRole.java:21]: Yeah, I will take a close look at opportunities for things like this when I get around to removing the plugin extension point for node roles.

File: docs/reference/migration/migrate_8_0/settings.asciidoc (hunk 1 above)
Comment (jasontedor) [reply docs/reference/migration/migrate_8_0/settings.asciidoc]: I pushed 2362e7cb13f88272373965c757d62c17cbed8b88.

File: docs/reference/migration/migrate_8_0/settings.asciidoc (hunk 2 above)
Comment (jasontedor) [reply docs/reference/migration/migrate_8_0/settings.asciidoc]: I pushed 2362e7cb13f88272373965c757d62c17cbed8b88.
